python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -414,26 +539,49 @@ def reconcile(
     asset_graph = repository_def.asset_graph
 
     (
-        assets_to_reconcile,
+        asset_partitions_to_reconcile,
         newly_materialized_root_asset_keys,
+        newly_materialized_root_partitions_by_asset_key,
         latest_storage_id,
-    ) = determine_assets_to_reconcile(
+    ) = determine_asset_partitions_to_reconcile(
         instance_queryer=instance_queryer,
         asset_graph=asset_graph,
         cursor=cursor,
         target_asset_selection=asset_selection,
     )
 
-    if assets_to_reconcile:
-        run_requests = [RunRequest(asset_selection=list(assets_to_reconcile), tags=run_tags)]
-    else:
-        run_requests = []
+    assets_to_reconcile_by_partitions_def_partition_key: Mapping[
+        Tuple[Optional[PartitionsDefinition], Optional[str]], Set[AssetKey]
+    ] = defaultdict(set)
+
+    for asset_partition in asset_partitions_to_reconcile:
+        assets_to_reconcile_by_partitions_def_partition_key[
+            asset_graph.get_partitions_def(asset_partition.asset_key), asset_partition.partition_key
+        ].add(asset_partition.asset_key)
+
+    run_requests = []
+
+    for (
+        _,
+        partition_key,
+    ), asset_keys in assets_to_reconcile_by_partitions_def_partition_key.items():
+        tags = {**(run_tags or {})}
+        if partition_key is not None:
+            tags[PARTITION_NAME_TAG] = partition_key
```

Comment:
I think we'll need to assign different tags for multi-dimensional partitions.

I can add this in a future PR though--probably worth aggregating all of the partition tag logic into one function to avoid this logic being duplicated over many different call sites

Reply:
Ah, makes sense. Sounds good.